modules/virt-configuring-huge-pages-for-vms.adoc

```diff
@@ -0,0 +1,54 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/advanced_vm_management/virt-using-huge-pages-with-vms.adoc
+
+[id="virt-configuring-huge-pages-for-vms_{context}"]
+= Configuring huge pages for virtual machines
+
+You can configure virtual machines to use pre-allocated huge pages by including the
+`memory.hugepages.pageSize` and memory request parameters in your virtual machine configuration.
+
+The memory request must be divisible by the page size. For example, you can not request `500Mi` memory with a page size of `1Gi`.
+
+[NOTE]
+====
+The memory layouts of the host and the guest OS are unrelated.
+Hugepages requested in the virtual machine manifest apply to QEMU.
+Hugepages inside the guest can only be configured based on the amount of available memory of the virtual machine instance.
+====
+
+If you edit a running virtual machine, the virtual machine must be rebooted for the changes to take effect.
+
+.Prerequisites
+
+* Nodes must have pre-allocated huge pages configured.
```

Comment:
Yeah, in the built doc the assembly prereq is quite a ways from this procedure because of how large the concept is. Also if this topic was re-used I wanted to make sure the pre-req wasn't lost.